tests/suites/test_suite_psa_crypto.data

```diff
@@ -188,65 +188,65 @@ PSA import/export EC secp224r1 key pair: good
 depends_on:MBEDTLS_PK_PARSE_C:MBEDTLS_PK_WRITE_C:MBEDTLS_ECP_C:MBEDTLS_ECP_DP_SECP224R1_ENABLED
 import_export:"6849f97d1066f6997759637c7e3899464cee3ec7ac970653a0be0742":PSA_KEY_TYPE_ECC_KEYPAIR(PSA_ECC_CURVE_SECP224R1):PSA_ALG_ECDSA_ANY:PSA_KEY_USAGE_EXPORT:224:0:PSA_SUCCESS:1
 
-PSA import/export-public EC secp224r1: good
+PSA import/export-public EC secp224r1: good XXX not sure why this was bit string not octet string
```

Comment:
???

Reply:
This is something I noticed when reading the ASN.1 and reading SEC 1 (http://www.secg.org/sec1-v2.pdf). We previously were using `BIT_STRING` (`0x03`) instead of `OCTET_STRING` (`0x04`) per the standard.

```
ECPoint ::= OCTET STRING
```

This might be worth fling an inter-op bug about unless we know some reason using `BIT_STRING` is correct. It's not an issue for PSA anymore, since we won't use an ASN.1-formatted key, but is still a potential issue for Mbed TLS.

Reply:
Oh, I see what you mean. I'm quite confident that there is no interoperability bug since Mbed TLS is able to interact with the other TLS implementations out there. The bug is partly on me for using the name `ECPoint` for something that is not what the applicable specifications call `ECPoint`, and I would say partly on SEC1 for having confusing language.

[RFC 5280](https://tools.ietf.org/html/rfc5280#section-4.1) defines SubjectPublicKeyInfo as containing a subjectPublicKey which is always a BIT STRING. The content of this BIT STRING depends on the key type. [RFC 3279](https://tools.ietf.org/html/rfc3279#section-2.3) defines the content of a subjectPublicKey for some common key types. For an RSA or DSA or DH key, the content of the BIT STRING is an ASN.1 SEQUENCE. For an EC key, SEC 1 §C.3 says (the same language appears in [RFC 3279 §2.3.5](https://tools.ietf.org/html/rfc3279#section-2.3.5)):

> The elliptic curve public key (an ECPoint which is an OCTET STRING) is mapped to a subjectPublicKey (a BIT STRING) as follows: the most significant bit of the OCTET STRING becomes the most significant bit of the BIT STRING, and the least significant bit of the OCTET STRING becomes the least significant bit of the BIT STRING.

What you do here is to take the _content_ of the ECPoint and stuff it into a BIT STRING. You don't stuff the OCTET STRING (`04` + length + content) into the BIT STRING. By coincidence, the content of the BIT STRING happens to start with `04` in all the data we manipulate because that's the prefix for an uncompressed point representation (SEC1 §2.3.3 action 3.3). You can argue that SEC1 is ambiguous, and I'd agree with that, but everybody including the de facto standard openssl stuffs an EC point representation into a BIT STRING and not an ECPoint into a BIT STRING.

I filed a spec bug to go through all uses of “ECPoint” and fix them.